core/integration/module_test.go

```diff
@@ -3190,6 +3201,14 @@ class Status {
 
 @object()
 export class Test {
+  @func()
+  status: Status
+
+  // FIXME: this should be Status.Inactive instead of "INACTIVE"
+  constructor(status: Status = "INACTIVE") {
```

Comment:
@TomChv this is kinda a nasty hack, but this works to make sure that this actually works :tada:

If it looks good to you, I'll open a follow-up issue, and can try tackling that when I have a bit more time.

Reply:
LGTM, resolving reference by their accessor it's a complex issue, related to TS SDK only, I'll fix that later!

Reply:
Spun out to https://github.com/dagger/dagger/issues/8152!